packages/components/Button/src/Button.styling.ts

```diff
@@ -32,22 +32,60 @@ export const stylingSettings: UseStylingOptions<ButtonProps, ButtonSlotProps, Bu
   tokens: [defaultButtonTokens, defaultButtonFontTokens, defaultButtonColorTokens, buttonName],
   states: buttonStates,
   slotProps: {
-    root: buildProps(
-      (tokens: ButtonTokens, theme: Theme) => ({
-        style: {
-          display: 'flex',
-          alignItems: 'center',
-          flexDirection: 'row',
-          alignSelf: 'flex-start',
-          justifyContent: 'center',
-          width: tokens.width,
-          backgroundColor: tokens.backgroundColor,
-          ...borderStyles.from(tokens, theme),
-          ...layoutStyles.from(tokens, theme),
-        },
-      }),
-      ['backgroundColor', 'width', ...borderStyles.keys, ...layoutStyles.keys],
-    ),
+    ...Platform.select({
```

Comment:
I think we should have a `Button.styling.android.ts` file. There's a precedent for that. It'll also get picked up by jest tests, where `Platform.OS / .select` won't without mocking.

Reply:
I take it back. For styling files it's probably fine. For cleaner code I think I'd rather we do a spread:

```
...(Platform.OS === 'android' && {android_ripple: ...})
```

Reply:
I think Platform.Select might be cleaner here since there is also some code being added for "not Android", this would have to be placed in a separate if statement otherwise

Reply:
What "not Android code" is there? As far as I can tell, the `root` slot of both Android and default are the same, minus an extra `ripple_config`. It should be safe to add the ripple config to other Platforms because that's a prop from RN Core. Rather than duplicating the Root slot twice (which could lead to bugs where engineers update the root on win32/macOS/iOS but not Android), I'd rather we just have one root slot and _maybe_ conditionally create the `rippleContainer` slot per platform. Even if we just _always_ created the slotProps for ripple container on all platforms, I don't think that would be too bad.

Reply:
Alright, will make the change to have only one root